packages/devextreme/js/__internal/grids/grid_core/virtual_scrolling/m_virtual_scrolling.ts

```diff
@@ -1613,6 +1613,10 @@ export const rowsView = (Base: ModuleType<RowsView>) => class VirtualScrollingRo
       .viewportItemSize(rowHeight);
 
     if (isVirtualMode(this) || gridCoreUtils.isVirtualRowRendering(this)) {
+      if (this._dataController.isEmpty()) {
+        return;
+      }
```

Comment:
Minor suggestion: move the expression inside if statement into a variable
```suggestion
      const skipRenderEmptyRows = this._dataController.isEmpty();
      
      if (skipRenderEmptyRows) {
        return;
      }
```